### ISSUE TITLE
Bulk access to documents using bulk input

### ISSUE THREAD

**Issue by kltm** [geneontology/amigo#69]
The bulk download of annotations with gene product input is an occasionally used feature of AmiGO 1.x that I suspect might be missed.

The easiest implementation would be a series of queries to the GOlr server, with the results hashed on the perl side as we go to get rid of dupes.

**Comment by cmungall** [comment]
Some comments on current status of bulk search (not sure if this is redundant with some of the extensive info above).

## GO ID use case

 * This is v useful when you have a set of GO IDs
 * It's not completely clear what box to click on the right. "GO class (direct) (annotation_class)" is pretty opaque. And it doesn't do what is expected. It appears to use the closure (which is actually what most users want, even if they don't have the language to express it)
 * Arriving at a list of GO IDs is obviously a challenge for many users, be awesome when this is hooked up, e.g. shopping carts
 * Entering in GO term labels is fun, but again most people don't know the strings by memory, and getting the required list may be a challenge for some. Hooking up to the SG annotator would be awesome here. 

## Gene use case

 * there are checkboxes for bioentity label and bioentity name, but none for the ID. Most will have IDs
 * Guaranteed that all kinds of mad IDs will be entered. If we are serious about this functionality we should be sure we have similar behavior for term enrichment (the gene use case can be seen as a degenerate case of TE)

## Genercity

It's great that this is driven by the schema metadata... from a CS perspective. But I think the experience will be alienating for a user here. Perhaps a more fruitful long term approach would be grebe with the ability to enter lists?

## Breaking resolution into a separate step

There is no way for the user to see which subset of entered IDs resolve. Really resolution and bulk query are separate concerns. There are many cases where we might want to plug in the resolution part (e.g. TE) and many places where we want to include the bulk query part.

Text annotation can be seen as a special case of resolution. For example, try the default query here:
https://monarchinitiative.org/annotate/text

You should end up with a box that says "35 terms found" and a button to search with these 35.

Another scenario is where the user enters IDs one at a time, e.g.
https://monarchinitiative.org/analyze/phenotypes/

Both text annotation (for terms or genes) and autocomplete based list building are equally useful for GO

**Comment by kltm** [comment]
(The plain ID is just "bioentity"--it's being driven off of the metadata for other displays.)

There is a bit of question of scope here. There is the unarguably power tool aspect here--this obviously needs inside knowledge--but it does possibly fill a niche as is. It was relatively easy to create given that it is bootstrapped from the metadata, but it is obviously not particularly useful to most users (hence it never graduating from labs).

I think it would be good to determine if the current tool, or something similar, has any prospects (it was made for an initial narrow use case) and then try and work out from there. If a grebe-ified version would be more useful to users, we should probably just branch it off and try again now that the troubling parts work better.

Once label and ID resolution is brought in, I think we should scrape and try for a new tool--there is just so much packed into that there is unlikely much that could be salvaged from here.

The cart chugging along again would be so nice...

**Comment by kltm** [comment]
(The plain ID is just "bioentity"--it's being driven off of the metadata for other displays.)

There is a bit of question of scope here. There is the unarguably power tool aspect here--this obviously needs inside knowledge--but it does possibly fill a niche as is. It was relatively easy to create given that it is bootstrapped from the metadata, but it is obviously not particularly useful to most users (hence it never graduating from labs).

I think it would be good to determine if the current tool, or something similar, has any prospects (it was made for an initial narrow use case) and then try and work out from there. If a grebe-ified version would be more useful to users, we should probably just branch it off and try again now that the troubling parts work better.

Once label and ID resolution is brought in, I think we should scrape and try for a new tool--there is just so much packed into that there is unlikely much that could be salvaged from here.

The cart chugging along again would be so nice...

**Comment by kltm** [comment]
Still interest here: http://jira.geneontology.org/browse/GO-1224
We really should just go ahead and add this at some point, or something; @mcourtot does the tool as it stands look usable to you for some use cases?
 http://tomodachi.berkeleybop.org/amigo/bulk_search/annotation
 http://tomodachi.berkeleybop.org/amigo/bulk_search/ontology
 http://tomodachi.berkeleybop.org/amigo/bulk_search/bioentity

**Comment by mcourtot** [comment]
Hi Seth, 

I had a quick look and it is quite unclear what the categories are (in the attached screenshot); what is the difference between the first and fourth cheekboxes for example?
![screen shot 2016-05-24 at 11 38 34](https://cloud.githubusercontent.com/assets/2214135/15501259/79590980-21a4-11e6-8857-df50585c758a.png)

Also re download, it would be good to have 'download as GAF, GPAD, txt or else instead of the columns choice maybe? This is a bit overwhelming.

Finally, when searching for GO IDs, QuickGO allows to search for the exact terms, or for their terms and descendants (the latter being the default, typically desired, behaviour), while AmiGO seems to search only the exact ID. 

In short, the functionality seems to be pretty much there, but the UI is a bit too complex IMHO.

Cheers,
Melanie

**Comment by kltm** [comment]
There could be some simplification added (e.g. GAF download on the annotation bulk download), and certainly better explanatory text for the different fields, but fundamentally the reason we essentially get these bulk widgets and pages for "free" is that they run right along the software patterns that are used underneath.

**Comment by vanaukenk** [comment]
There is still interest in this feature:
http://jira.geneontology.org/browse/GO-1277

**Comment by kltm** [comment]
talking to @hattrill , for the annotation search we should defulat to checked boxes for going from gene names/ids to terms.

**Comment by kltm** [comment]
Gene/product/bioentity: http://amigo.geneontology.org/amigo/bulk_search/bioentity
Annotation: http://amigo.geneontology.org/amigo/bulk_search/annotation
Ontology term/class: http://amigo.geneontology.org/amigo/bulk_search/ontology

**Comment by pgaudet** [comment]
Nice ! Can we add links from the 'Search' menu?

**Comment by pgaudet** [comment]
Although you might want to remove 'This should not be displayed (bioentity_internal_id)' ... 
(although it's also in the AmiGO download)

**Comment by kltm** [comment]
@pgaudet This feature is still in development--this is still the same as shown at the meeting.

**Comment by ValWood** [comment]
did not work with  input
SPAC23D3.14c
SPCC63.02c
SPAC4F10.02
SPAC3H5.04
SPBC359.03c
SPBC2D10.18
SPAC3F10.11c
SPBC359.05

**Comment by kltm** [comment]
Hm. Looking at just "SPBC359.05", I couldn't find it in the "normal" AmiGO either. Is it possible we currently don't have those identifiers? Could you give me an ID for "SPBC359.05"?

**Comment by ValWood** [comment]
yes, thinking about it I didn't see yeast either when I was looking for C. elegans, so it was probably the same issue...

**Comment by ukemi** [comment]
Testing: Gene/product/bioentity: http://amigo.geneontology.org/amigo/bulk_search/bioentity

The search should be made case insensitive. Using the Gene/product (bioentity_label) and searching on Trib3 should return the human TRIB3 gene as well as mouse and rat.

To make things clearer, I would relabel the choices:
Gene/product (bioentity)----Gene Identifier using a prefix:identifier format (eg MGI:MGI:1345675 or PomBase:SPCC63.02c)
Gene/product (bioentity_label)---- Gene Symbol (eg Trib3)
Gene/product name (bioentity_name)---- Gene or protein name
This should not be displayed (bioentity_internal_id)------- get rid of this choice
Synonyms (synonym)----- Synonyms
Involved in (isa_partof_closure_label)----- 'GO term' without genes that are annotated to 'regulation of GO term' I tried 'heart development' and 'limb development' no results. Also tried GO identifiers with no results returned.
Inferred annotation (regulates_closure)---- 'GO term' including genes annotated to 'regulation of GO term'. I tried GO:0043405 and got back a list of genes that don't make sense.
Inferred annotation (regulates_closure_label)---- What is the difference between this and the choice above this?
PANTHER family (panther_family)---- This is useful, but doesn't seem to be working
PANTHER family (panther_family_label)-----This is useful, but doesn't seem to be working
Organism (taxon_label)----- How useful is this? If it works, won't it return all the genes for a given organism?
